fsh/ig-data/input/pagecontent/index.md

```diff
@@ -67,161 +94,280 @@ An example call from an app to the EHR client is presented below.
-| `responseToMessageId` | CONDITIONAL  | The `messageId` of the received message that this message is in response to.  MAY NOT exist in [unsolicited response messages](#unsolicited-responses). |
-| `status`              | REQUIRED     | An HTTP-response code ([reference](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)). |
-| `targetOrigin`        | RECOMMENDED  | The contents of `smart_messaging_origin` from the launch context parameter; applications SHOULD refrain from using `"*"` for this. |
+| Property              | Optionality  | Type   | Description |
+| --------------------- | ------------ | ------ | ----------- |
+| `messageId`           | REQUIRED     | ID     | A unique ID for this message generated by the application. |
+| `responseToMessageId` | REQUIRED  | ID     | The `messageId` of the received message that this message is in response to.|
+| `payload`             | REQUIRED     | object | The message content as specified by the `messageType` of the request message.  See [below](#response-payload). |
 {:.grid}
 
-In addition, the response can hold additional fields as specified by the message category of the original message.
+In addition to sending Web Messaging Responses to the app, the EHR MAY also send "unsolicited" requests to the app, omitting  a `responseToMessageId` property. The semantics of such requests are are implementation-specific.
+#### Response Payload
+The response message `payload` properties will vary based on the request `messageType`.  See message types below for details.
+
+#### Response Target Origin
+It is assumed that the EHR already knows the proper `targetOrigin` to use in its
+call to [`window.postMessage`] because it has demonstrated that it can SMART
+launch the app at a known SMART launch URL.  The SMART launch URL SHALL be
+prefixed with the proper value to use for the app `targetOrigin`, and the launch
+URL SHALL not redirect users to a different origin after launch.
+
+#### Detailed Example Response
+In a more detailed example response, the EHR may send one return message like:
 
-#### Example Response
-In response, the EHR may send one return message like:
 ```js
-appWindow.postMessage({
-  "messageId": '<some new guid>',
-  "messageTye": '<see message types below>',
-  "responseToMessageId": '<original guid>',
-  "payload": {       // as for FHIR's Bundle.entry.response
-    "status": 404,   // HTTP response codes, for simplicity, consistency
-    "location": "",  // could be relative for scratchpad-like stuff?
-    "outcome":  "",  // include the resource as created
+window.addEventListener("message", function(event) {
+  if (event.origin != "<the app's expected origin>") {
+    return;  // Ignore unknown origins.
   }
-}, targetOrigin)
+
+  const messageType = event.data.messageType;
+
+  // The response payload is modeled after Bundle.entry.response.
+  // See: https://www.hl7.org/fhir/bundle-definitions.html#Bundle.entry.response
+  var location = "Example/123";
```

Comment:
Can we inline these below? The variables aren't used elsewhere (and if we do keep declarations, should be `const`).

Reply:
Yeah, I was trying to suggest that the `TODO: handle message here` code below the declarations might update those variables, but I can see that it's a bit awkward to do it that way.  Inline is clearer.